packages/template-retail-react-app/scripts/build-translations.js

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env node
+/*
+ * Copyright (c) 2023, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+/* eslint @typescript-eslint/no-var-requires: "off" */
+
+/**
+ * This script will extract messages from base template and extended app and have the output inside extended app
+ * If a file is overridden, it won't extract messages from that file in the base template
+ */
+const {exec} = require('child_process')
+const fs = require('fs')
+const path = require('path')
+const packagePath = path.join(process.cwd(), '.', 'package.json')
+const pkgJSON = JSON.parse(fs.readFileSync(packagePath))
+
+const getAllFilesByExtensions = (dirPath, arrayOfFiles = [], extensions = []) => {
+    const files = fs.readdirSync(dirPath)
+
+    files.forEach(function (file) {
+        if (fs.statSync(path.join(dirPath, file)).isDirectory()) {
+            arrayOfFiles = getAllFilesByExtensions(
+                path.join(dirPath, file),
+                arrayOfFiles,
+                extensions
+            )
+        } else {
+            arrayOfFiles.push(path.join(dirPath, file))
+        }
+    })
+    if (extensions.length) {
+        return arrayOfFiles.filter((filePath) => {
+            const getExtension = path.extname(filePath).replace('.', '')
+            return extensions.includes(getExtension)
+        })
+    }
+
+    return arrayOfFiles
+}
+// look for overridden files
+try {
+    const isBaseTemplate = !!pkgJSON.ccExtensibility.extendable
+    if (isBaseTemplate) {
+        const command =
+            "formatjs extract 'app/**/*.{js,jsx}' --out-file app/translations/en-US.json --id-interpolation-pattern [sha512:contenthash:base64:6]"
+        exec(command, (err, stdout) => {
+            if (err) {
+                console.error(err)
+            }
+            console.log('stout', stdout)
+        })
+        process.exit(1)
+    }
+    const overridesDir = path.join(process.cwd(), '.', pkgJSON.ccExtensibility?.overridesDir)
+    const fileNames = getAllFilesByExtensions(
+        path.join(overridesDir, 'app'),
+        [],
+        ['js', 'jsx', 'ts', 'tsx']
+    )
+    // files that will be ignored in translation extraction
+    const filesToIgnore = fileNames
```

Comment:
Are this absolute paths? If they are I imagine that this list will get large pretty quick. We probably need to document somewhere that this is a limitation, especially on windows. Partners do some crazy things.. so you never know what will happen.

Reply:
No, they are actually relative path. e.g `--ignore ./node_modules/retail-react-app/app/components/_app/index.jsx` Limit command line length on windows are around 32,768 characters. the script will only ignore the file if it is overridden, and exists in the base template, then we will have this ignore string generated.
Maybe we should warn if it exceeds some limit for windows, If they do exceed the limit, they can't use our script and  have to generate extract messages on their own. 🤔

Reply:
@alexvuong alternatively, we can shorten the ignore argument in this way:

1. For all the files to ignore, append the file name with a new extension: `component.jsx.ignored`
2. So then we can tell the CLI to ignore all files with the `.ignored` file extension
3. Then finally restore the above files back to their original file extension

Reply:
that is a brilliant idea!